### PR TITLE
[drape] Don't change location icon when the map is dragged

### PR DESCRIPTION
**Pull request by rtsisyk**
Keep PendingLocation mode when the user touch or drag the map.

## Test Plan

Please see a new test case added to docs/ANDROID_LOCATION_TEST.md. iPhone hasn't been tested—please validate also.

### CODE REVIEW

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14

File: drape_frontend/my_position_controller.cpp [hunk 1]
```diff
@@ -459,16 +448,7 @@ void MyPositionController::OnLocationUpdate(location::GpsInfo const & info, bool
 
   if (!m_isPositionAssigned)
   {
-    // If the position was never assigned, the new mode will be the desired one except next cases:
     location::EMyPositionMode newMode = m_desiredInitMode;
-    if (m_mode == location::NotFollowNoPosition)
```
Comment (vng) [inline drape_frontend/my_position_controller.cpp:464]: Put ASSERT(m_mode != location::NotFollowNoPosition, ());
Just in case ..

**Review by vng — COMMENTED** [review]
Remove ```m_mode == location::NotFollowNoPosition```
in MyPositionController::CheckIsWaitingForLocation() ?

File: drape_frontend/my_position_controller.cpp (hunk 1 above)
Comment (rtsisyk) [reply drape_frontend/my_position_controller.cpp:464]: I am not sure here, but let's try and see if it fires.

Comment (rtsisyk) [reply drape_frontend/my_position_controller.cpp:464]: Added

**Comment by rtsisyk** [comment]
macOS tests are failing because of #6472

**Review by vng — APPROVED** [review]

**Comment by pastk** [comment]
A great and long-awaited fix! Many thanks!!

**Pull request merged**